### PR TITLE
Fixes #41 Add pagination with numerical values to News searches

### LINKED ISSUE #41: Add pagination with numerical values to News searches
When terms are searched on the news section of the website there is a 'next' button to go through search items, but no numerical values on the bottom to quickly see the amount of search results or to jump through pages of results.

https://coderdojo.com/news/

### PR DESCRIPTION
**Pull request by rossoneill**

### CODE REVIEW

File: templates/index.twig [hunk 1]
```diff
@@ -3,13 +3,24 @@
 {% block content %}
 <div class="col-md-12">
 	{% include "article-list.twig" with {"wordCount": 100} %}
-	<div class="pagination-buttons">
+<div class="text-center">
+	<ul class="pagination pagination-centered">
 		{% if pagination.prev %}
-		<a href="{{pagination.prev.link}}" class="prev {{pagination.prev.link|length ? '' : 'invisible'}} pull-left"><button class="btn btn-default">Previous</button></a>
+		<a href="{{pagination.prev.link}}" class="prev {{pagination.prev.link|length ? '' : 'invisible'}}"><button class="btn btn-default">Previous</button></a>
```
Comment (DanielBrierton) [inline templates/index.twig]: `a` tag not allowed directly in `ul`. Should wrap this in an `li`

File: templates/index.twig [hunk 2]
```diff
@@ -3,13 +3,24 @@
...
+						{% if page.link %}
+								<a href="{{page.link}}" class="btn btn-default">{{page.title}}</a>
+						{% else %}
+								<span class="btn btn-default">{{page.title}}</span>
+						{% endif %}
+				</li>
+		{% endfor %}
 		{% if pagination.next %}
-		<a href="{{pagination.next.link}}" class="next {{pagination.next.link|length ? '' : 'invisible'}} pull-right"><button class="btn btn-default">Next</button></a>
+		<a href="{{pagination.next.link}}" class="next {{pagination.next.link|length ? '' : 'invisible'}}"><button class="btn btn-default">Next</button></a>
```
Comment (DanielBrierton) [inline templates/index.twig]: same as before, `a` tag not allowed in `ul`

Comment (DanielBrierton) [inline templates/index.twig]: Also, you shouldn't have a `<button>` inside the `<a>`. You can just put "Next" directly inside the `<a>`

File: templates/index.twig [hunk 3]
```diff
@@ -3,13 +3,24 @@
...
+<div class="text-center">
+	<ul class="pagination pagination-centered">
 		{% if pagination.prev %}
-		<a href="{{pagination.prev.link}}" class="prev {{pagination.prev.link|length ? '' : 'invisible'}} pull-left"><button class="btn btn-default">Previous</button></a>
+		<a href="{{pagination.prev.link}}" class="prev {{pagination.prev.link|length ? '' : 'invisible'}}"><button class="btn btn-default">Previous</button></a>
 		{% endif %}
+		{% for page in pagination.pages %}
+				<li>
+						{% if page.link %}
+								<a href="{{page.link}}" class="btn btn-default">{{page.title}}</a>
```
Comment (DanielBrierton) [inline templates/index.twig]: the `class="btn btn-default"` is causing rounded corners on each item when there shouldn't be. Remove the classes, and it'll look better.

**Comment by rossoneill** [comment]
I believe this has been done for all including news items.

File: templates/index.twig [hunk 4]
```diff
@@ -3,13 +3,28 @@
...
 		{% if pagination.prev %}
-		<a href="{{pagination.prev.link}}" class="prev {{pagination.prev.link|length ? '' : 'invisible'}} pull-left"><button class="btn btn-default">Previous</button></a>
+			<li>
+					<a class="btn" href="{{pagination.prev.link}}" class="prev {{pagination.prev.link|length ? '' : 'invisible'}}">Previous</a>
+			</li>
 		{% endif %}
+		{% for page in pagination.pages %}
+				<li>
+						{% if page.link %}
+								<li><a href="{{page.link}}" class="btn">{{page.title}}</a></li>
```
Comment (DanielBrierton) [inline templates/index.twig]: The `btn` class needs to be removed from here too, as it's giving the links a rounded corner, which we don't want.

File: templates/index.twig [hunk 5]
```diff
@@ -3,13 +3,28 @@
...
+						{% else %}
+								<li><span class="btn">{{page.title}}</span></li>
```
Comment (DanielBrierton) [inline templates/index.twig]: Same again here, `btn` needs to be removed.

File: templates/index.twig [hunk 6]
```diff
@@ -3,13 +3,28 @@
 {% block content %}
 <div class="col-md-12">
 	{% include "article-list.twig" with {"wordCount": 100} %}
-	<div class="pagination-buttons">
+<div class="text-center">
+	<ul class="pagination pagination-centered">
 		{% if pagination.prev %}
-		<a href="{{pagination.prev.link}}" class="prev {{pagination.prev.link|length ? '' : 'invisible'}} pull-left"><button class="btn btn-default">Previous</button></a>
+			<li>
+					<a class="btn" href="{{pagination.prev.link}}" class="prev {{pagination.prev.link|length ? '' : 'invisible'}}">Previous</a>
```
Comment (DanielBrierton) [inline templates/index.twig]: Indented too far. This is also the case for lines 14-20

File: templates/index.twig [hunk 7]
```diff
@@ -3,13 +3,28 @@
...
+								<li><a href="{{page.link}}" class="btn">{{page.title}}</a></li>
+						{% else %}
+								<li><span class="btn">{{page.title}}</span></li>
+						{% endif %}
+				</li>
+		{% endfor %}
 		{% if pagination.next %}
-		<a href="{{pagination.next.link}}" class="next {{pagination.next.link|length ? '' : 'invisible'}} pull-right"><button class="btn btn-default">Next</button></a>
+		<li>
+		<a class="btn" href="{{pagination.next.link}}" class="next {{pagination.next.link|length ? '' : 'invisible'}}">Next</a>
```
Comment (DanielBrierton) [inline templates/index.twig]: Indent

File: templates/index.twig [hunk 8]
```diff
@@ -3,13 +3,28 @@
 {% block content %}
 <div class="col-md-12">
 	{% include "article-list.twig" with {"wordCount": 100} %}
-	<div class="pagination-buttons">
+<div class="text-center">
```
Comment (DanielBrierton) [inline templates/index.twig]: Whole block should be indented from here down to the matching closing `div`.

File: views/page-news.twig [hunk 9]
```diff
@@ -5,20 +5,33 @@
...
-          {% if pagination.next %}
-          <a href="{{pagination.next.link}}" class="next {{pagination.next.link|length ? '' : 'invisible'}} pull-right"><button class="btn btn-default">Next</button></a>
-          {% endif %}
-        </div>
-      </div>
-    </article>
+    <h1>All News</h1>
+    <div class="col-md-9">
+    	{% include "article-list.twig" with {"wordCount": 50} %}
+    <div class="text-center">
```
Comment (DanielBrierton) [inline views/page-news.twig]: `div` should be indented from here to closing `div`

**Pull request merged**